app/presenters/salary_question_presenter.rb

```diff
@@ -5,15 +5,16 @@ def response_label(value)
     format_salary(SmartAnswer::Salary.new(value))
   end
 
-  def amount
-    return response_for_current_question unless response_for_current_question.is_a? Hash
+  def parsed_response
+    return {} if current_response.blank?
+    return current_response if current_response.is_a?(Hash)
 
-    response_for_current_question[:amount]
-  end
-
-  def period
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:period]
+    salary = @node.parse_input(current_response)
+    {
+      amount: salary.amount,
+      period: salary.period,
+    }
+  rescue SmartAnswer::InvalidResponse
+    {}
```

Comment:
I have the same concerns as with DateQuestionPresenter. How come we don't use the underlying object return by `parse_input` and are we suppressing errors shown to the user?

Reply:
Hopefully addressed in previous comments.